spec/classes/stub_spec.rb

```diff
@@ -0,0 +1,275 @@
+require 'spec_helper'
+
+describe 'bin/stub' do
+  let(:subject_path) { File.expand_path('stub', "#{File.dirname(__FILE__)}/../../bin") }
+  let(:stub_call_pathname) { instance_double('Pathname') }
```

Comment:
I've tried to keep the things that are verified as implementation non-specific as possible. This worked well for exit codes, ARGV, stderr, and stderr. However, it doesn't work so well for how the stub accesses its config, log and output files.